spec/views/hq/orphans/_filters_spec.rb

```diff
@@ -12,6 +10,21 @@
   end
 
   describe 'has form fields' do
+    before :each do
+      allow(Orphan).to receive_message_chain(:distinct, :pluck).with(:health_status)
+                            .and_return(["health_status1", "health_status2", orphans_filters[:health_status].to_s])
+      allow(Province).to receive_message_chain(:distinct, :pluck).with(:name, :code)
+                            .and_return([["province1", "1"],["Selected Province", orphans_filters[:province_code].to_s]])
+      allow(Orphan).to receive_message_chain(:distinct, :pluck).with(:priority)
+                            .and_return(["priority1","priority2", orphans_filters[:priority].to_s])
+      allow(Orphan).to receive_message_chain(:distinct, :pluck).with(:city)
+                            .and_return(["city1","city2", orphans_filters[:original_address_city].to_s])
+      allow(Orphan).to receive_message_chain(:distinct, :pluck).with(:sponsorship_status)
+                            .and_return( (0..(Orphan.sponsorship_statuses.size-1)).to_a )
+      allow(Orphan).to receive_message_chain(:distinct, :pluck).with(:status)
+                            .and_return( (0..(Orphan.statuses.size-1)).to_a )
+      allow(Partner).to receive_message_chain(:all_names).and_return(["partner1", "partner2", orphans_filters[:orphan_list_partner_name]])
+    end
```

Comment:
stubbing all in a before block is cleaner. 
using `show_me_rendered` you can check that all fields are correctly filled